#      **Trabajo Nº 1. Datasets potenciales para el proyecto de Data Science.**








Enlace a archivo guardado en drive.

In [ ]:
from google.colab import drive
drive.mount("/content/drive", force_remount=True)

Mounted at /content/drive


Llamo a las librerias potenciales a utilizar.

In [ ]:
import pandas as pd
import numpy as np

***Carga del correspondiente dataset Nº1.***





Introduccion: Este dataset es acerca de Covid19 donde tenemos datos diferenciasdos tanto por continente como por pais donde informa datos importantes por dia de la cantidad de casos nuevos, activos, criticos y curados tambien como la cantidad de testeos y muertes.

In [ ]:
datos = pd.read_csv('/content/drive/MyDrive/Data science- Datasets/Covid19DB/covid19.csv',index_col=0,sep=';')
datos.head(5)

index continent      country  population        day  \
0      0      Asia  Afghanistan  40418964.0  17/3/2022   
1      1      Asia  Afghanistan  40416563.0  17/3/2022   
2      2      Asia  Afghanistan  40414161.0  16/3/2022   
3      3      Asia  Afghanistan  40411760.0  15/3/2022   
4      4      Asia  Afghanistan  40409359.0  14/3/2022   

                        time  cases_new  cases_active  cases_critical  \
0  2022-03-17T20:15:03+00:00      128.0        9698.0          1124.0   
1  2022-03-17T08:00:04+00:00      126.0        9858.0          1124.0   
2  2022-03-16T08:00:04+00:00      108.0       10008.0          1124.0   
3  2022-03-15T09:15:03+00:00      180.0       10257.0          1124.0   
4  2022-03-14T09:15:03+00:00      118.0       10129.0          1124.0   

   cases_recovered  cases_1M_pop  cases_total  deaths_new  deaths_1M_pop  \
0         159569.0        4377.0       176918         NaN          189.0   
1         159234.0        4373.0       176743         2.0          189.0   
2         158917.0        4369.0       176571         1.0          189.0   
3         158507.0        4365.0       176409         2.0          189.0   
4         158429.0        4360.0       176201         1.0          189.0   

   deaths_total  tests_1M_pop  tests_total  
0        7651.0       22496.0     909246.0  
1        7651.0       22463.0     907896.0  
2        7646.0       22434.0     906639.0  
3        7645.0       22410.0     905633.0  
4        7643.0       22372.0     904055.0

Informacion acerca de dataset.

In [ ]:
datos.shape

(232220, 17)

Podemos visualizar 232220 filas y 17 columnas

In [ ]:
datos.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 232220 entries, 0 to 232220
Data columns (total 17 columns):
 #   Column           Non-Null Count   Dtype  
---  ------           --------------   -----  
 0   index            232220 non-null  int64  
 1   continent        231156 non-null  object 
 2   country          232220 non-null  object 
 3   population       208219 non-null  float64
 4   day              232220 non-null  object 
 5   time             232220 non-null  object 
 6   cases_new        201739 non-null  float64
 7   cases_active     229447 non-null  float64
 8   cases_critical   187608 non-null  float64
 9   cases_recovered  228498 non-null  float64
 10  cases_1M_pop     232198 non-null  float64
 11  cases_total      232220 non-null  int64  
 12  deaths_new       150552 non-null  float64
 13  deaths_1M_pop    204717 non-null  float64
 14  deaths_total     226542 non-null  float64
 15  tests_1M_pop     204255 non-null  float64
 16  tests_total      221566 non-null  floa

Podemos visualizar una tabla que indica el indice de numeros de columnas a la izquierda, nombre de todas la columnas en la seguna columna vertical, el numero de datos non-null y ademas el tipo de dato en la ultima columna de la derecha. Ademas en la parte final la memoria utilizada.

In [ ]:
datos.describe()

index    population      cases_new  cases_active  \
count  232220.000000  2.082190e+05  201739.000000  2.294470e+05   
mean     1902.392895  9.478735e+07   10176.502436  7.478784e+05   
std      4254.582970  2.354371e+08   35110.684237  2.726760e+06   
min         0.000000  8.010000e+02       1.000000 -1.268120e+06   
25%       280.000000  2.973206e+06      66.000000  1.019000e+03   
50%       597.000000  1.132082e+07     577.000000  1.112400e+04   
75%      1068.000000  6.033582e+07    4302.000000  9.103700e+04   
max     24379.000000  1.439324e+09  848502.000000  2.917711e+07   

       cases_critical  cases_recovered   cases_1M_pop   cases_total  \
count    1.876080e+05     2.284980e+05  232198.000000  2.322200e+05   
mean     2.856007e+03     2.653763e+06   43383.411329  3.421956e+06   
std      8.196924e+03     7.685923e+06   59896.446723  1.017720e+07   
min      1.000000e+00     1.000000e+00       0.020000  1.000000e+00   
25%      1.700000e+01     7.140000e+03    2273.000000  1.139500e+04   
50%      1.540000e+02     9.788900e+04   17760.000000  1.405535e+05   
75%      1.402000e+03     8.251970e+05   66992.000000  1.046677e+06   
max      2.279900e+06     5.663943e+07  696299.000000  8.130004e+07   

          deaths_new  deaths_1M_pop   deaths_total  tests_1M_pop   tests_total  
count  150552.000000  204717.000000  226542.000000  2.042550e+05  2.215660e+05  
mean      180.097441     742.003158   62942.635547  8.415106e+05  4.923285e+07  
std       427.608430     862.832704  162014.223727  1.695037e+06  1.423794e+08  
min       -22.000000       0.010000       1.000000  4.000000e+00  1.000000e+01  
25%         4.000000      78.000000     174.000000  7.828300e+04  2.479310e+05  
50%        19.000000     411.000000    2512.000000  2.925840e+05  2.123570e+06  
75%       121.000000    1153.000000   21872.000000  9.564425e+05  1.530876e+07  
max      6138.000000    6270.000000  995111.000000  2.169926e+07  9.690993e+08

Podemos ver lista de de columnas correscpondientes.
Donde vemos cantidad de valores no nulos, la media, la desviacion, el valor minimo, lo que hay debajo del 25%, 50%, 75% y ademas el numero maximo.


In [ ]:
datos.index

Int64Index([     0,      1,      2,      3,      4,      5,      6,      7,
                 8,      9,
            ...
            232211, 232212, 232213, 232214, 232215, 232216, 232217, 232218,
            232219, 232220],
           dtype='int64', length=232220)

Podemos visualizar el numero de filas que tenemos la cual es 232220

In [ ]:
datos.columns

Index(['index', 'continent', 'country', 'population', 'day', 'time',
       'cases_new', 'cases_active', 'cases_critical', 'cases_recovered',
       'cases_1M_pop', 'cases_total', 'deaths_new', 'deaths_1M_pop',
       'deaths_total', 'tests_1M_pop', 'tests_total'],
      dtype='object')

Tenemos el nombre de cada columna y su totalidad (17)

In [ ]:
columnas = datos.columns
columnas

Index(['index', 'continent', 'country', 'population', 'day', 'time',
       'cases_new', 'cases_active', 'cases_critical', 'cases_recovered',
       'cases_1M_pop', 'cases_total', 'deaths_new', 'deaths_1M_pop',
       'deaths_total', 'tests_1M_pop', 'tests_total'],
      dtype='object')

Guardo la info de columnas.

In [ ]:
type(columnas)

pandas.core.indexes.base.Index

Podemos visualizar el tipo de dato de las respectivas columnas.

In [ ]:
datos.tail(5)

index continent   country  population        day  \
232216    876    Africa  Zimbabwe         NaN  15/4/2020   
232217    877    Africa  Zimbabwe         NaN  13/4/2020   
232218    878    Africa  Zimbabwe         NaN  12/4/2020   
232219    879    Africa  Zimbabwe         NaN  11/4/2020   
232220    880    Africa  Zimbabwe         NaN  10/4/2020   

                             time  cases_new  cases_active  cases_critical  \
232216  2020-04-15T03:00:05+00:00        3.0          14.0             NaN   
232217  2020-04-13T20:45:08+00:00        NaN          11.0             NaN   
232218  2020-04-12T22:15:06+00:00        1.0          11.0             NaN   
232219  2020-04-11T22:00:16+00:00        2.0          10.0             NaN   
232220  2020-04-10T20:15:04+00:00        NaN           8.0             NaN   

        cases_recovered  cases_1M_pop  cases_total  deaths_new  deaths_1M_pop  \
232216              NaN           1.0           17         NaN            NaN   
232217              NaN           0.9           14         NaN            NaN   
232218              NaN           0.9           14         NaN            NaN   
232219              NaN           0.9           13         NaN            NaN   
232220              NaN           0.7           11         NaN            NaN   

        deaths_total  tests_1M_pop  tests_total  
232216           3.0           NaN        604.0  
232217           3.0           NaN        563.0  
232218           3.0           NaN        547.0  
232219           3.0           NaN        438.0  
232220           3.0           NaN        438.0

Podemos visualizar las ultimas 5 filas.

In [ ]:
datos.country

0         Afghanistan
1         Afghanistan
2         Afghanistan
3         Afghanistan
4         Afghanistan
             ...     
232216       Zimbabwe
232217       Zimbabwe
232218       Zimbabwe
232219       Zimbabwe
232220       Zimbabwe
Name: country, Length: 232220, dtype: object

Obtencion de valores de una columa en concreto. Se tomo de ejemplo la columna country.Podemos visualizar los primeros 4 y ultimos 5 datos.

In [ ]:
datos.iloc[:,0]

0           0
1           1
2           2
3           3
4           4
         ... 
232216    876
232217    877
232218    878
232219    879
232220    880
Name: index, Length: 232220, dtype: int64

Obtener los datos por indice - iloc

In [ ]:
datos.loc[datos.country=='Aregtina']

Empty DataFrame
Columns: [index, continent, country, population, day, time, cases_new, cases_active, cases_critical, cases_recovered, cases_1M_pop, cases_total, deaths_new, deaths_1M_pop, deaths_total, tests_1M_pop, tests_total]
Index: []

En esta seccion deberiamos ver la la lista con la condicion del pais asentado. Esto es productivo para poder visualizar el pais en interes a visualizar los datos, o tambien se puede realizar el mismo filtro por 'continent' si se desa visualizar el continente.

In [ ]:
columnas_nulos = datos.columns[datos.isnull().any()]
datos[columnas_nulos].isnull().sum()

continent           1064
population         24001
cases_new          30481
cases_active        2773
cases_critical     44612
cases_recovered     3722
cases_1M_pop          22
deaths_new         81668
deaths_1M_pop      27503
deaths_total        5678
tests_1M_pop       27965
tests_total        10654
dtype: int64

Podemos visualizar la cantidad de valores nulos hay en cada columna.

In [ ]:
datos.isnull().sum()

index                  0
continent           1064
country                0
population         24001
day                    0
time                   0
cases_new          30481
cases_active        2773
cases_critical     44612
cases_recovered     3722
cases_1M_pop          22
cases_total            0
deaths_new         81668
deaths_1M_pop      27503
deaths_total        5678
tests_1M_pop       27965
tests_total        10654
dtype: int64

El mismo recuento pero agregando ademas los que no tienen nulos.

In [ ]:
datos.isnull().any()

index              False
continent           True
country            False
population          True
day                False
time               False
cases_new           True
cases_active        True
cases_critical      True
cases_recovered     True
cases_1M_pop        True
cases_total        False
deaths_new          True
deaths_1M_pop       True
deaths_total        True
tests_1M_pop        True
tests_total         True
dtype: bool

Indica si hay datos nulos en booleano.

In [ ]:
datos_sin_nulos = datos.dropna(how='any')

Borramos los los nulos. 

In [ ]:
datos_sin_nulos.isnull().sum()

index              0
continent          0
country            0
population         0
day                0
time               0
cases_new          0
cases_active       0
cases_critical     0
cases_recovered    0
cases_1M_pop       0
cases_total        0
deaths_new         0
deaths_1M_pop      0
deaths_total       0
tests_1M_pop       0
tests_total        0
dtype: int64

Se comprueba que exactamente no contenga nulos.

In [ ]:
datos.shape

(232220, 17)

In [ ]:
datos_sin_nulos.shape

(117295, 17)

Comparativa de disminucion de filas sacando los nulos.

In [ ]:
datos.country.unique()

array(['Afghanistan', 'Albania', 'Algeria', 'Andorra', 'Angola',
       'Anguilla', 'Antigua-and-Barbuda', 'Argentina', 'Armenia', 'Aruba',
       'Australia', 'Austria', 'Azerbaijan', 'Bahamas', 'Bahrain',
       'Bangladesh', 'Barbados', 'Belarus', 'Belgium', 'Belize', 'Benin',
       'Bermuda', 'Bhutan', 'Bolivia', 'Bosnia-and-Herzegovina',
       'Botswana', 'Brazil', 'British-Virgin-Islands', 'Brunei',
       'Bulgaria', 'Burkina-Faso', 'Burundi', 'Cabo-Verde', 'Cambodia',
       'Cameroon', 'Canada', 'CAR', 'Caribbean-Netherlands',
       'Cayman-Islands', 'Chad', 'Channel-Islands', 'Chile', 'China',
       'Colombia', 'Comoros', 'Congo', 'Cook-Islands', 'Costa-Rica',
       'Croatia', 'Cuba', 'Cyprus', 'Czechia', 'Denmark',
       'Diamond-Princess', 'Diamond-Princess-', 'Djibouti', 'Dominica',
       'Dominican-Republic', 'DRC', 'Ecuador', 'Egypt', 'El-Salvador',
       'Equatorial-Guinea', 'Eritrea', 'Estonia', 'Eswatini', 'Ethiopia',
       'Faeroe-Islands', 'Falkland-Islands

Visualizamos los valores unicos para previsualizar los paises a los cuales se listaron en dicho dataset.

In [ ]:
datos.continent.unique()

array(['Asia', '38807744', 'Europe', '2878189', 'Africa', '77249', nan,
       'North-America', '97824', 'South-America', '45141794', '2962544',
       '106708', 'Oceania', 'Australia/Oceania', '25460983', '8999806',
       '10127290', '1693017', '164475123', '9449712', '11583189', '62306',
       '770494', '11651907', '3283308', '212364444', '30205', '436933',
       '6954790', '555195', '16688259', '26454069', '37699156', '65620',
       '173654', '19094896', '50811863', '4108356', '1206224', '10706473',
       '5789609', '986088', '10833701', '89174452', '17607301', '6482018',
       '1396368', '1326423', '114568201', '48839', '3466', '5539635',
       '65255252', '297602', '2218477', '3990118', '83749935', '13082852',
       '1961573', '9883522', '7489099', '9663390', '340958', '1378233611',
       '273144953', '83850893', '4930547', '84975', '8637437', '60472892',
       '10189909', '18747133', '4262156', '6509752', '7261510', '1888690',
       '6829201', '2726772', '624613', '323

Visualizamos los valores unicos para previsualizar los continentes a los cuales se listaron en dicho dataset.

In [ ]:
agrupados=datos.groupby('country').deaths_total.count()
agrupados

country
Afghanistan          958
Albania              908
Algeria              870
Andorra              716
Angola               782
                    ... 
Wallis-and-Futuna    231
Western-Sahara         6
Yemen                646
Zambia               965
Zimbabwe             881
Name: deaths_total, Length: 229, dtype: int64

Podriamos visualizar una lista de paises contando cantidad de muertes totales.

In [ ]:
agrupados=datos.groupby('country').deaths_total.min()
agrupados

country
Afghanistan           1.0
Albania               2.0
Algeria              15.0
Andorra               1.0
Angola                2.0
                     ... 
Wallis-and-Futuna     1.0
Western-Sahara        1.0
Yemen                 2.0
Zambia                1.0
Zimbabwe              3.0
Name: deaths_total, Length: 229, dtype: float64

Podemos visualizar las muertes minimas por pais.

In [ ]:
datos.groupby(['country','deaths_total']).deaths_total.agg([min,max])

min     max
country     deaths_total                
Afghanistan 1.0              1.0     1.0
            2.0              2.0     2.0
            4.0              4.0     4.0
            6.0              6.0     6.0
            7.0              7.0     7.0
...                          ...     ...
Zimbabwe    5412.0        5412.0  5412.0
            5414.0        5414.0  5414.0
            5416.0        5416.0  5416.0
            5417.0        5417.0  5417.0
            5418.0        5418.0  5418.0

[108361 rows x 2 columns]

Podemos diferenciar la lista con el minimo y maximo de muertes.

***Carga del correspondiente dataset Nº2.***

Introduccion:Este dataset corresponde al ultimo censo realizado en la argentina, donde en las columnas vemos el diferencial de rango de edad con su respectivo indicador de genero y en la filas visualizamos el diferencial por comuna/sector. Entiendo que no cumple con la cantidad de filas demandadas pero me interesa la informacion de dicho dataset.

In [ ]:
datos = pd.read_csv('/content/drive/MyDrive/Data science- Datasets/Poblacion ArgentinaDb/poblacion_ar_dpto.csv',index_col=0,sep=';')
datos.head(5)

name  0-4_m  0-4_f  5-9_m  5-9_f  10-14_m  10-14_f  15-19_m  \
in1                                                                      
2001  Comuna 01   5926   5743   5079   5070     4973     4940     5737   
2002  Comuna 02   3206   3014   2879   2817     2765     2721     3786   
2003  Comuna 03   5078   4811   4723   4504     4570     4416     5023   
2004  Comuna 04   8013   7576   7613   7439     7696     7410     7880   
2005  Comuna 05   4594   4412   4469   4190     4198     4097     4696   

      15-19_f  20-24_m  ...  75-79_m  75-79_f  80-84_m  80-84_f  85-89_m  \
in1                     ...                                                
2001     5845     9721  ...     2040     3457     1564     2948      821   
2002     4453     7457  ...     2103     3598     1529     3436      951   
2003     5188     7945  ...     2060     3874     1613     3458      803   
2004     7948     9079  ...     2182     3504     1381     3098      752   
2005     4694     6489  ...     2152     3785     1639     3570      833   

      85-89_f  90-94_m  90-94_f  >95_m  >95_f  
in1                                            
2001     1990      285      888     66    301  
2002     2399      347     1164     81    440  
2003     2073      242      830     62    298  
2004     1674      177      658     37    176  
2005     2299      237      981     58    292  

[5 rows x 41 columns]

In [ ]:
datos.shape

(527, 41)

Podemos visualizar 527 filas y 41 columnas.

In [ ]:
datos.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 527 entries, 2001 to 94028
Data columns (total 41 columns):
 #   Column   Non-Null Count  Dtype 
---  ------   --------------  ----- 
 0   name     527 non-null    object
 1   0-4_m    527 non-null    int64 
 2   0-4_f    527 non-null    int64 
 3   5-9_m    527 non-null    int64 
 4   5-9_f    527 non-null    int64 
 5   10-14_m  527 non-null    int64 
 6   10-14_f  527 non-null    int64 
 7   15-19_m  527 non-null    int64 
 8   15-19_f  527 non-null    int64 
 9   20-24_m  527 non-null    int64 
 10  20-24_f  527 non-null    int64 
 11  25-29_m  527 non-null    int64 
 12  25-29_f  527 non-null    int64 
 13  30-34_m  527 non-null    int64 
 14  30-34_f  527 non-null    int64 
 15  35-39_m  527 non-null    int64 
 16  35-39_f  527 non-null    int64 
 17  40-44_m  527 non-null    int64 
 18  40-44_f  527 non-null    int64 
 19  45-49_m  527 non-null    int64 
 20  45-49_f  527 non-null    int64 
 21  50-54_m  527 non-null    int64 
 2

Podemos visualizar una tabla que indica el indice de numeros de columnas a la izquierda, nombre de todas la columnas en la seguna columna vertical, el numero de datos non-null y ademas el tipo de dato en la ultima columna de la derecha. Ademas en la parte final la memoria utilizada.

In [ ]:
datos.describe()

0-4_m         0-4_f         5-9_m         5-9_f       10-14_m  \
count    527.000000    527.000000    527.000000    527.000000    527.000000   
mean    3221.958254   3111.347249   3259.491461   3156.483871   3376.417457   
std     6274.017369   6069.628933   6118.009449   5925.933746   6273.431059   
min        0.000000      0.000000      0.000000      0.000000      0.000000   
25%      484.500000    464.000000    512.500000    493.500000    547.500000   
50%     1260.000000   1210.000000   1326.000000   1269.000000   1391.000000   
75%     3222.000000   3113.500000   3444.000000   3283.500000   3610.500000   
max    81900.000000  79331.000000  78587.000000  76355.000000  79938.000000   

            10-14_f       15-19_m       15-19_f       20-24_m       20-24_f  \
count    527.000000    527.000000    527.000000    527.000000    527.000000   
mean    3271.487666   3387.212524   3333.977230   3128.000000   3134.142315   
std     6120.539060   6511.039973   6473.927454   6462.707384   6613.791680   
min        0.000000      0.000000      0.000000      0.000000      0.000000   
25%      528.000000    529.000000    514.500000    399.500000    386.500000   
50%     1345.000000   1329.000000   1232.000000   1016.000000   1006.000000   
75%     3461.500000   3559.000000   3419.000000   2862.000000   2894.500000   
max    78925.000000  81812.000000  80971.000000  76270.000000  78600.000000   

       ...       75-79_m       75-79_f      80-84_m       80-84_f  \
count  ...    527.000000    527.000000   527.000000    527.000000   
mean   ...    610.020873    911.155598   380.918406    692.927894   
std    ...   1183.875693   1898.057397   749.731870   1477.619923   
min    ...      0.000000      0.000000     0.000000      0.000000   
25%    ...     80.000000     96.000000    50.500000     68.000000   
50%    ...    198.000000    257.000000   123.000000    185.000000   
75%    ...    546.000000    728.000000   328.500000    544.500000   
max    ...  11453.000000  18189.000000  7159.000000  14809.000000   

           85-89_m      85-89_f     90-94_m      90-94_f       >95_m  \
count   527.000000   527.000000  527.000000   527.000000  527.000000   
mean    176.182163   389.922201   50.425047   144.658444   10.413662   
std     352.451136   845.439396   96.809118   306.485005   19.956694   
min       0.000000     0.000000    0.000000     0.000000    0.000000   
25%      23.000000    35.000000    7.000000    14.000000    1.000000   
50%      52.000000    98.000000   17.000000    38.000000    3.000000   
75%     154.000000   304.000000   46.000000   112.500000    9.000000   
max    3337.000000  8690.000000  902.000000  3046.000000  192.000000   

            >95_f  
count  527.000000  
mean    40.757116  
std     88.650475  
min      0.000000  
25%      4.000000  
50%     11.000000  
75%     30.000000  
max    864.000000  

[8 rows x 40 columns]

Podemos ver lista de de columnas correscpondientes. Donde vemos cantidad de valores no nulos, la media, la desviacion, el valor minimo, lo que hay debajo del 25%, 50%, 75% y ademas el numero maximo.

In [ ]:
datos.index

Int64Index([ 2001,  2002,  2003,  2004,  2005,  2006,  2007,  2008,  2009,
             2010,
            ...
            90084, 90091, 90098, 90105, 90112, 90119, 94007, 94014, 94021,
            94028],
           dtype='int64', name='in1', length=527)

In [ ]:
datos.columns

Index(['name', '0-4_m', '0-4_f', '5-9_m', '5-9_f', '10-14_m', '10-14_f',
       '15-19_m', '15-19_f', '20-24_m', '20-24_f', '25-29_m', '25-29_f',
       '30-34_m', '30-34_f', '35-39_m', '35-39_f', '40-44_m', '40-44_f',
       '45-49_m', '45-49_f', '50-54_m', '50-54_f', '55-59_m', '55-59_f',
       '60-64_m', '60-64_f', '65-69_m', '65-69_f', '70-74_m', '70-74_f',
       '75-79_m', '75-79_f', '80-84_m', '80-84_f', '85-89_m', '85-89_f',
       '90-94_m', '90-94_f', '>95_m', '>95_f'],
      dtype='object')

Se confirma con dichos datos lo estipulado anteriormente para mostrar las 527 filas y 47 columnas.

In [ ]:
columnas1 = datos.columns
columnas1


Index(['name', '0-4_m', '0-4_f', '5-9_m', '5-9_f', '10-14_m', '10-14_f',
       '15-19_m', '15-19_f', '20-24_m', '20-24_f', '25-29_m', '25-29_f',
       '30-34_m', '30-34_f', '35-39_m', '35-39_f', '40-44_m', '40-44_f',
       '45-49_m', '45-49_f', '50-54_m', '50-54_f', '55-59_m', '55-59_f',
       '60-64_m', '60-64_f', '65-69_m', '65-69_f', '70-74_m', '70-74_f',
       '75-79_m', '75-79_f', '80-84_m', '80-84_f', '85-89_m', '85-89_f',
       '90-94_m', '90-94_f', '>95_m', '>95_f'],
      dtype='object')

Guardo la info de columnas.

In [ ]:
type(columnas1)

pandas.core.indexes.base.Index

Podemos visualizar el tipo de dato de las respectivas columnas.

In [ ]:
datos.tail(5)

name  0-4_m  0-4_f  5-9_m  5-9_f  10-14_m  10-14_f  \
in1                                                                            
90119              Yerba Buena   3182   3123   3666   3647     3972     3684   
94007               R�o Grande   3379   3064   3396   3216     3379     3198   
94014                  Ushuaia   2545   2490   2591   2389     2650     2533   
94021  Islas del Atl�ntico sur      0      0      0      0        0        0   
94028      Ant�rtida Argentina      2      0      4      3        1        4   

       15-19_m  15-19_f  20-24_m  ...  75-79_m  75-79_f  80-84_m  80-84_f  \
in1                               ...                                       
90119     3812     3780     3263  ...      327      474      191      324   
94007     3373     3113     3249  ...      203      269       83      174   
94014     2582     2535     2326  ...      140      184       77      138   
94021        0        0        0  ...        0        0        0        0   
94028        2        0        3  ...        0        0        0        0   

       85-89_m  85-89_f  90-94_m  90-94_f  >95_m  >95_f  
in1                                                      
90119       82      175       22       53      4     22  
94007       46       77        7       25      3      5  
94014       30       71        5       22      2      4  
94021        0        0        0        0      0      0  
94028        0        0        0        0      0      0  

[5 rows x 41 columns]

Podemos visualizar las ultimas 5 filas.

In [ ]:
datos.name

in1
2001                   Comuna 01
2002                   Comuna 02
2003                   Comuna 03
2004                   Comuna 04
2005                   Comuna 05
                  ...           
90119                Yerba Buena
94007                 R�o Grande
94014                    Ushuaia
94021    Islas del Atl�ntico sur
94028        Ant�rtida Argentina
Name: name, Length: 527, dtype: object

Obtencion de valores de una columna en concreto. Se tomo de ejemplo la columna name(comunas/sector).Podemos visualizar los primeros 4 y ultimos 5 datos.

In [ ]:
datos.loc[datos.name=='Capital']

name  0-4_m  0-4_f  5-9_m  5-9_f  10-14_m  10-14_f  15-19_m  \
in1                                                                     
10049  Capital   6866   6845   7433   7332     7693     7827     7726   
14014  Capital  53168  51845  52248  50455    53314    52153    56820   
18021  Capital  14710  14238  15826  15307    16975    16501    18431   
42021  Capital   4294   4103   4264   4202     4413     4314     4575   
46014  Capital   8199   7710   8186   8113     8638     8421     9445   
50007  Capital   3878   3814   3206   3189     3395     3379     4032   
54028  Capital  14984  14515  15421  14893    16813    16521    16784   
66028  Capital  24305  23525  25015  24250    26120    25209    27104   
70028  Capital   4162   4034   4002   3847     3888     3735     4067   
86049  Capital  11319  11026  12244  12011    13770    13479    13309   
90084  Capital  22409  21973  23038  22449    24115    23502    24664   

       15-19_f  20-24_m  ...  75-79_m  75-79_f  80-84_m  80-84_f  85-89_m  \
in1                      ...                                                
10049     7792     6730  ...      857     1457      476     1002      200   
14014    57227    61671  ...    10090    16634     6205    12609     2684   
18021    18638    17200  ...     1829     3238     1089     2032      471   
42021     4586     4280  ...      810     1218      504      968      232   
46014     9638     8651  ...      857     1301      452      900      196   
50007     4146     5644  ...     1245     2179      938     1870      498   
54028    17151    13981  ...     1758     2613      991     1737      415   
66028    27427    23054  ...     3323     4826     1944     3233      891   
70028     4077     3997  ...     1155     2017      758     1636      392   
86049    13516    10954  ...     1441     2501      969     1803      402   
90084    24995    24605  ...     3845     6702     2387     4841     1052   

       85-89_f  90-94_m  90-94_f  >95_m  >95_f  
in1                                             
10049      526       68      197     15     58  
14014     6951      766     2461    128    703  
18021     1115      183      411     34    116  
42021      561       68      189      7     61  
46014      448       58      168     32     75  
50007     1093      151      429     27    105  
54028      811      132      324     29     96  
66028     1821      259      665     58    165  
70028      838       95      276     26     82  
86049      961      125      369     38    110  
90084     2517      272      848     70    251  

[11 rows x 41 columns]

En esta seccion deberiamos ver la la lista con la condicion de la comuna/sector asentado. Esto es productivo para poder visualizar sector en interes a visualizar los datos.

In [ ]:
columnas_nulos = datos.columns[datos.isnull().any()]
datos[columnas_nulos].isnull().sum()

Series([], dtype: float64)

Podemos visualizar la cantidad de valores nulos hay en cada columna. Donde casualmente visualizamos que el dataset no contiene valores nulos.

In [ ]:
datos.isnull().any()

name       False
0-4_m      False
0-4_f      False
5-9_m      False
5-9_f      False
10-14_m    False
10-14_f    False
15-19_m    False
15-19_f    False
20-24_m    False
20-24_f    False
25-29_m    False
25-29_f    False
30-34_m    False
30-34_f    False
35-39_m    False
35-39_f    False
40-44_m    False
40-44_f    False
45-49_m    False
45-49_f    False
50-54_m    False
50-54_f    False
55-59_m    False
55-59_f    False
60-64_m    False
60-64_f    False
65-69_m    False
65-69_f    False
70-74_m    False
70-74_f    False
75-79_m    False
75-79_f    False
80-84_m    False
80-84_f    False
85-89_m    False
85-89_f    False
90-94_m    False
90-94_f    False
>95_m      False
>95_f      False
dtype: bool

Confirmamos con metodo booleano lo asentado previamente.

In [ ]:
datos.name.unique()

array(['Comuna 01', 'Comuna 02', 'Comuna 03', 'Comuna 04', 'Comuna 05',
       'Comuna 06', 'Comuna 07', 'Comuna 08', 'Comuna 09', 'Comuna 10',
       'Comuna 11', 'Comuna 12', 'Comuna 13', 'Comuna 14', 'Comuna 15',
       'Almirante Brown', 'Avellaneda', 'Berazategui',
       'Esteban Echeverr�a', 'Ezeiza', 'Florencio Varela',
       'General San Mart�n', 'Hurlingham', 'Ituzaing�', 'Jos� C. Paz',
       'La Matanza', 'Lan�s', 'Lomas de Zamora', 'Malvinas Argentinas',
       'Merlo', 'Moreno', 'Mor�n', 'Quilmes', 'San Fernando',
       'San Isidro', 'San Miguel', 'Tigre', 'Tres de Febrero',
       'Vicente L�pez', 'Adolfo Alsina', 'Adolfo Gonz�les Chaves',
       'Alberti', 'Ayacucho', 'Azul', 'Bah�a Blanca', 'Balcarce',
       'Baradero', 'Arrecifes', 'Benito Ju�rez', 'Berisso', 'Bol�var',
       'Bragado', 'Brandsen', 'Campana', 'Ca�uelas', 'Capit�n Sarmiento',
       'Carlos Casares', 'Carlos Tejedor', 'Carmen de Areco', 'Castelli',
       'Col�n', 'Coronel de Marina L. Rosales', 'C

Podemos visualizar el listado de comunas/sectores asentados en el dataset.

***Carga del correspondiente dataset Nº3.***

Introduccion:Este dataset corresponde a los valores de rentas correspondiente a ubicaciones mas solicitadas de la india. Tenemos mucha informacion bastante interesante como el dia de posteo, el valor, el tamaño el tipo, cantidad de pisos, la localidad,ciudad, numero de baños y si es publicado por una agencia o por el dueño directo.

In [ ]:
datos = pd.read_csv('/content/drive/MyDrive/Data science- Datasets/House rent dataset/House_Rent_Dataset.csv',index_col=0,sep=';')
datos.head(5)

La carga de datos ha sido exitosa, podemos visualizar las primeros 5 filas con sus respectivas columnas.

In [ ]:
datos.shape

(4746, 11)

Podemos visualizar el numero de filas 4746 y columas 11

In [ ]:
datos.info()

<class 'pandas.core.frame.DataFrame'>
Index: 4746 entries, 18/05/2022 to 04/05/2022
Data columns (total 11 columns):
 #   Column             Non-Null Count  Dtype 
---  ------             --------------  ----- 
 0   BHK                4746 non-null   int64 
 1   Rent               4746 non-null   int64 
 2   Size               4746 non-null   int64 
 3   Floor              4746 non-null   object
 4   Area Type          4746 non-null   object
 5   Area Locality      4746 non-null   object
 6   City               4746 non-null   object
 7   Furnishing Status  4746 non-null   object
 8   Tenant Preferred   4746 non-null   object
 9   Bathroom           4746 non-null   int64 
 10  Point of Contact   4746 non-null   object
dtypes: int64(4), object(7)
memory usage: 444.9+ KB


Podemos visualizar una tabla que indica el indice de numeros de columnas a la izquierda, nombre de todas la columnas en la seguna columna vertical, el numero de datos non-null y ademas el tipo de dato en la ultima columna de la derecha. Ademas en la parte final la memoria utilizada.

In [ ]:
datos.describe()

BHK          Rent         Size     Bathroom
count  4746.000000  4.746000e+03  4746.000000  4746.000000
mean      2.083860  3.499345e+04   967.490729     1.965866
std       0.832256  7.810641e+04   634.202328     0.884532
min       1.000000  1.200000e+03    10.000000     1.000000
25%       2.000000  1.000000e+04   550.000000     1.000000
50%       2.000000  1.600000e+04   850.000000     2.000000
75%       3.000000  3.300000e+04  1200.000000     2.000000
max       6.000000  3.500000e+06  8000.000000    10.000000

Podemos ver lista de de columnas correscpondientes. Donde vemos cantidad de valores no nulos, la media, la desviacion, el valor minimo, lo que hay debajo del 25%, 50%, 75% y ademas el numero maximo.

In [ ]:
datos.index

Index(['18/05/2022', '13/05/2022', '16/05/2022', '04/07/2022', '09/05/2022',
       '29/04/2022', '21/06/2022', '21/06/2022', '07/06/2022', '20/06/2022',
       ...
       '28/06/2022', '07/07/2022', '06/07/2022', '06/07/2022', '02/06/2022',
       '18/05/2022', '15/05/2022', '10/07/2022', '06/07/2022', '04/05/2022'],
      dtype='object', name='Posted On', length=4746)

In [ ]:
datos.columns

Index(['BHK', 'Rent', 'Size', 'Floor', 'Area Type', 'Area Locality', 'City',
       'Furnishing Status', 'Tenant Preferred', 'Bathroom',
       'Point of Contact'],
      dtype='object')

Podemos confirmar el numero de filas con el numero de columnas.

In [ ]:
columnas3 = datos.columns
columnas3

Index(['BHK', 'Rent', 'Size', 'Floor', 'Area Type', 'Area Locality', 'City',
       'Furnishing Status', 'Tenant Preferred', 'Bathroom',
       'Point of Contact'],
      dtype='object')

Guardo la info de columnas.

In [ ]:
type(columnas3)

pandas.core.indexes.base.Index

Podemos visualizar el tipo de dato de las respectivas columnas.

In [ ]:
datos.tail(5)

BHK   Rent  Size         Floor    Area Type         Area Locality  \
Posted On                                                                       
18/05/2022    2  15000  1000    3 out of 5  Carpet Area          Bandam Kommu   
15/05/2022    3  29000  2000    1 out of 4   Super Area  Manikonda, Hyderabad   
10/07/2022    3  35000  1750    3 out of 5  Carpet Area  Himayath Nagar, NH 7   
06/07/2022    3  45000  1500  23 out of 34  Carpet Area            Gachibowli   
04/05/2022    2  15000  1000    4 out of 5  Carpet Area       Suchitra Circle   

                 City Furnishing Status  Tenant Preferred  Bathroom  \
Posted On                                                             
18/05/2022  Hyderabad    Semi-Furnished  Bachelors/Family         2   
15/05/2022  Hyderabad    Semi-Furnished  Bachelors/Family         3   
10/07/2022  Hyderabad    Semi-Furnished  Bachelors/Family         3   
06/07/2022  Hyderabad    Semi-Furnished            Family         2   
04/05/2022  Hyderabad       Unfurnished         Bachelors         2   

           Point of Contact  
Posted On                    
18/05/2022    Contact Owner  
15/05/2022    Contact Owner  
10/07/2022    Contact Agent  
06/07/2022    Contact Agent  
04/05/2022    Contact Owner

Podemos visualizar las ultimas 5 filas.

In [ ]:
datos.City

Posted On
18/05/2022      Kolkata
13/05/2022      Kolkata
16/05/2022      Kolkata
04/07/2022      Kolkata
09/05/2022      Kolkata
                ...    
18/05/2022    Hyderabad
15/05/2022    Hyderabad
10/07/2022    Hyderabad
06/07/2022    Hyderabad
04/05/2022    Hyderabad
Name: City, Length: 4746, dtype: object

Obtencion de valores de una columa en concreto. Se tomo de ejemplo la columna City.Podemos visualizar los primeros 4 y ultimos 5 datos.

In [ ]:
datos.loc[datos.City=='Delhi']

BHK   Rent  Size            Floor    Area Type  \
Posted On                                                    
24/06/2022    2  14000   800       1 out of 3   Super Area   
30/06/2022    1  20000   200  Ground out of 2   Super Area   
26/06/2022    3  80000  1800       1 out of 3  Carpet Area   
08/06/2022    1   9500   400       2 out of 4  Carpet Area   
08/06/2022    3  14000   600       2 out of 4  Carpet Area   
...         ...    ...   ...              ...          ...   
13/06/2022    2  31000  1200       3 out of 4   Super Area   
15/05/2022    1   9000   250  Ground out of 3   Super Area   
04/06/2022    2   9000   700       4 out of 5   Super Area   
01/07/2022    3  27000  1050  Ground out of 4  Carpet Area   
05/07/2022    2  12500   500       2 out of 4  Carpet Area   

                                                Area Locality   City  \
Posted On                                                              
24/06/2022                            Keshav Puram, Tri Nagar  Delhi   
30/06/2022                                          Hauz Khas  Delhi   
26/06/2022                                  South Extension 1  Delhi   
08/06/2022                                          Neb Sarai  Delhi   
08/06/2022                          kst chattarpur Apartments  Delhi   
...                                                       ...    ...   
13/06/2022                                          Pitampura  Delhi   
15/05/2022  Jai laxmi Apartments, Indraprastha Extension, ...  Delhi   
04/06/2022                                         Sant Nagar  Delhi   
01/07/2022                       Nalanda Apartment, Vikaspuri  Delhi   
05/07/2022                                        Laxmi Nagar  Delhi   

           Furnishing Status  Tenant Preferred  Bathroom Point of Contact  
Posted On                                                                  
24/06/2022    Semi-Furnished  Bachelors/Family         2    Contact Owner  
30/06/2022         Furnished         Bachelors         1    Contact Owner  
26/06/2022    Semi-Furnished  Bachelors/Family         3    Contact Owner  
08/06/2022       Unfurnished  Bachelors/Family         1    Contact Agent  
08/06/2022       Unfurnished         Bachelors         3    Contact Agent  
...                      ...               ...       ...              ...  
13/06/2022    Semi-Furnished  Bachelors/Family         2    Contact Owner  
15/05/2022    Semi-Furnished  Bachelors/Family         1    Contact Owner  
04/06/2022    Semi-Furnished  Bachelors/Family         1    Contact Owner  
01/07/2022       Unfurnished            Family         2    Contact Agent  
05/07/2022    Semi-Furnished  Bachelors/Family         2    Contact Agent  

[605 rows x 11 columns]

En esta seccion deberiamos ver la la lista con la condicion del pais asentado. Esto es productivo para poder visualizar sector de la india en interes a visualizar los datos, o tambien se puede realizar el mismo filtro por Localidad si se desea visualizar lo dicho.

In [ ]:
columnas_nulos = datos.columns[datos.isnull().any()]
datos[columnas_nulos].isnull().sum()

Series([], dtype: float64)

Podemos visualizar la cantidad de valores nulos hay en cada columna. Donde casualmente visualizamos que el dataset no contiene valores nulos.

In [ ]:
datos.isnull().any()

BHK                  False
Rent                 False
Size                 False
Floor                False
Area Type            False
Area Locality        False
City                 False
Furnishing Status    False
Tenant Preferred     False
Bathroom             False
Point of Contact     False
dtype: bool

Confirmamos con metodo booleano lo asentado previamente.

In [ ]:
datos.City.unique()

array(['Kolkata', 'Mumbai', 'Bangalore', 'Delhi', 'Chennai', 'Hyderabad'],
      dtype=object)

Podemos visualizar las distintivas ciudades publicadas en el dataset.

***Carga del correspondiente dataset Nº4.***

Introduccion: Agregare un Dataset mas debido a que uno de los visualizados previamente no cumplia con la cantidad de filas correspondientes. Por lo tanto, presento este Dataset de Telecom, una importante compañia de telefonica la cual el dataset secciona a los usuarios a partir de su localidad informacion varios datos redundantes como cantidad de llamadas, mensajes, consumos en lineas generales, etc.

In [ ]:
datos = pd.read_csv('/content/drive/MyDrive/Data science- Datasets/Telecom Dataset/telecom_churn.csv',index_col=0,sep=';')
datos.head(5)

account length  area code phone number international plan  \
state                                                              
KS                128        415     382-4657                 no   
OH                107        415     371-7191                 no   
NJ                137        415     358-1921                 no   
OH                 84        408     375-9999                yes   
OK                 75        415     330-6626                yes   

      voice mail plan  number vmail messages  total day minutes  \
state                                                             
KS                yes                     25              265.1   
OH                yes                     26              161.6   
NJ                 no                      0              243.4   
OH                 no                      0              299.4   
OK                 no                      0              166.7   

       total day calls  total day charge  total eve minutes  total eve calls  \
state                                                                          
KS                 110             45.07              197.4               99   
OH                 123             27.47              195.5              103   
NJ                 114             41.38              121.2              110   
OH                  71             50.90               61.9               88   
OK                 113             28.34              148.3              122   

       total eve charge  total night minutes  total night calls  \
state                                                             
KS                16.78                244.7                 91   
OH                16.62                254.4                103   
NJ                10.30                162.6                104   
OH                 5.26                196.9                 89   
OK                12.61                186.9                121   

       total night charge  total intl minutes  total intl calls  \
state                                                             
KS                  11.01                10.0                 3   
OH                  11.45                13.7                 3   
NJ                   7.32                12.2                 5   
OH                   8.86                 6.6                 7   
OK                   8.41                10.1                 3   

       total intl charge  customer service calls  churn  
state                                                    
KS                  2.70                       1  False  
OH                  3.70                       1  False  
NJ                  3.29                       0  False  
OH                  1.78                       2  False  
OK                  2.73                       3  False

La carga de datos ha sido exitosa.

In [ ]:
datos.shape

(3333, 20)

Podemos visualizar que el dataset contiene un total de 3333 filas y 20 columnas

In [ ]:
datos.info()

<class 'pandas.core.frame.DataFrame'>
Index: 3333 entries, KS to TN
Data columns (total 20 columns):
 #   Column                  Non-Null Count  Dtype  
---  ------                  --------------  -----  
 0   account length          3333 non-null   int64  
 1   area code               3333 non-null   int64  
 2   phone number            3333 non-null   object 
 3   international plan      3333 non-null   object 
 4   voice mail plan         3333 non-null   object 
 5   number vmail messages   3333 non-null   int64  
 6   total day minutes       3333 non-null   float64
 7   total day calls         3333 non-null   int64  
 8   total day charge        3333 non-null   float64
 9   total eve minutes       3333 non-null   float64
 10  total eve calls         3333 non-null   int64  
 11  total eve charge        3333 non-null   float64
 12  total night minutes     3333 non-null   float64
 13  total night calls       3333 non-null   int64  
 14  total night charge      3333 non-null   float6

Podemos visualizar una tabla que indica el indice de numeros de columnas a la izquierda, nombre de todas la columnas en la seguna columna vertical, el numero de datos non-null y ademas el tipo de dato en la ultima columna de la derecha. Ademas en la parte final la memoria utilizada.

In [ ]:
datos.describe()

account length    area code  number vmail messages  total day minutes  \
count     3333.000000  3333.000000            3333.000000        3333.000000   
mean       101.064806   437.182418               8.099010         179.775098   
std         39.822106    42.371290              13.688365          54.467389   
min          1.000000   408.000000               0.000000           0.000000   
25%         74.000000   408.000000               0.000000         143.700000   
50%        101.000000   415.000000               0.000000         179.400000   
75%        127.000000   510.000000              20.000000         216.400000   
max        243.000000   510.000000              51.000000         350.800000   

       total day calls  total day charge  total eve minutes  total eve calls  \
count      3333.000000       3333.000000        3333.000000      3333.000000   
mean        100.435644         30.562307         200.980348       100.114311   
std          20.069084          9.259435          50.713844        19.922625   
min           0.000000          0.000000           0.000000         0.000000   
25%          87.000000         24.430000         166.600000        87.000000   
50%         101.000000         30.500000         201.400000       100.000000   
75%         114.000000         36.790000         235.300000       114.000000   
max         165.000000         59.640000         363.700000       170.000000   

       total eve charge  total night minutes  total night calls  \
count       3333.000000          3333.000000        3333.000000   
mean          17.083540           200.872037         100.107711   
std            4.310668            50.573847          19.568609   
min            0.000000            23.200000          33.000000   
25%           14.160000           167.000000          87.000000   
50%           17.120000           201.200000         100.000000   
75%           20.000000           235.300000         113.000000   
max           30.910000           395.000000         175.000000   

       total night charge  total intl minutes  total intl calls  \
count         3333.000000         3333.000000       3333.000000   
mean             9.039325           10.237294          4.479448   
std              2.275873            2.791840          2.461214   
min              1.040000            0.000000          0.000000   
25%              7.520000            8.500000          3.000000   
50%              9.050000           10.300000          4.000000   
75%             10.590000           12.100000          6.000000   
max             17.770000           20.000000         20.000000   

       total intl charge  customer service calls  
count        3333.000000             3333.000000  
mean            2.764581                1.562856  
std             0.753773                1.315491  
min             0.000000                0.000000  
25%             2.300000                1.000000  
50%             2.780000                1.000000  
75%             3.270000                2.000000  
max             5.400000                9.000000

Podemos ver lista de de columnas correscpondientes. Donde vemos cantidad de valores no nulos, la media, la desviacion, el valor minimo, lo que hay debajo del 25%, 50%, 75% y ademas el numero maximo.

In [ ]:
datos.index

Index(['KS', 'OH', 'NJ', 'OH', 'OK', 'AL', 'MA', 'MO', 'LA', 'WV',
       ...
       'IN', 'WV', 'OH', 'OH', 'SC', 'AZ', 'WV', 'RI', 'CT', 'TN'],
      dtype='object', name='state', length=3333)

In [ ]:
datos.columns

Index(['account length', 'area code', 'phone number', 'international plan',
       'voice mail plan', 'number vmail messages', 'total day minutes',
       'total day calls', 'total day charge', 'total eve minutes',
       'total eve calls', 'total eve charge', 'total night minutes',
       'total night calls', 'total night charge', 'total intl minutes',
       'total intl calls', 'total intl charge', 'customer service calls',
       'churn'],
      dtype='object')

Podemos confirmar el numero de filas con el numero de columnas.

In [ ]:
datos.tail(5)

account length  area code phone number international plan  \
state                                                              
AZ                192        415     414-4276                 no   
WV                 68        415     370-3271                 no   
RI                 28        510     328-8230                 no   
CT                184        510     364-6381                yes   
TN                 74        415     400-4344                 no   

      voice mail plan  number vmail messages  total day minutes  \
state                                                             
AZ                yes                     36              156.2   
WV                 no                      0              231.1   
RI                 no                      0              180.8   
CT                 no                      0              213.8   
TN                yes                     25              234.4   

       total day calls  total day charge  total eve minutes  total eve calls  \
state                                                                          
AZ                  77             26.55              215.5              126   
WV                  57             39.29              153.4               55   
RI                 109             30.74              288.8               58   
CT                 105             36.35              159.6               84   
TN                 113             39.85              265.9               82   

       total eve charge  total night minutes  total night calls  \
state                                                             
AZ                18.32                279.1                 83   
WV                13.04                191.3                123   
RI                24.55                191.9                 91   
CT                13.57                139.2                137   
TN                22.60                241.4                 77   

       total night charge  total intl minutes  total intl calls  \
state                                                             
AZ                  12.56                 9.9                 6   
WV                   8.61                 9.6                 4   
RI                   8.64                14.1                 6   
CT                   6.26                 5.0                10   
TN                  10.86                13.7                 4   

       total intl charge  customer service calls  churn  
state                                                    
AZ                  2.67                       2  False  
WV                  2.59                       3  False  
RI                  3.81                       2  False  
CT                  1.35                       2  False  
TN                  3.70                       0  False

Podemos visualizar las ultimas 5 filas correspondientes al dataset.

In [ ]:
datos.churn

state
KS    False
OH    False
NJ    False
OH    False
OK    False
      ...  
AZ    False
WV    False
RI    False
CT    False
TN    False
Name: churn, Length: 3333, dtype: bool

Obtencion de valores de una columa en concreto. Se tomo de ejemplo la columna Churn ya que vtuve dificultad al tomar otra columna por como esta escrito el titulo de las columnas..Podemos visualizar los primeros 4 y ultimos 5 datos.


In [ ]:
columnas_nulos = datos.columns[datos.isnull().any()]
datos[columnas_nulos].isnull().sum()

Series([], dtype: float64)

Podemos visualizar la cantidad de valores nulos hay en cada columna. Donde casualmente visualizamos que el dataset no contiene valores nulos.

In [ ]:
datos.isnull().any()

account length            False
area code                 False
phone number              False
international plan        False
voice mail plan           False
number vmail messages     False
total day minutes         False
total day calls           False
total day charge          False
total eve minutes         False
total eve calls           False
total eve charge          False
total night minutes       False
total night calls         False
total night charge        False
total intl minutes        False
total intl calls          False
total intl charge         False
customer service calls    False
churn                     False
dtype: bool

Confirmamos con metodo booleano lo asentado previamente.